Themes update

AlexanderBrevig opened the pull request:
This is a collection of changes made to force themes to handle in a predictable way. It also enforces a response to enabling features (like `cursorline`) so users who want these features will still see them even if they change themes.

I've tried my best to stick to the spirit of the theme, but some themes relied so much on terminal colors that I just had to guess and base some assumptions from other editors with similar themes.

Status from original authors (in terms of approvals)
- [x] snazzy @nrdxp 
- [ ] pop-dark @workingj 
- [x] penumbra @vlmutolo 
- [ ] noctis_bordo @jharrilim 
- [x] ingrid @intarga 
- [x] flatwhite @AlexanderBrevig 
- [x] doom_acario_dark @Rinnray 
- [x] dark_plus @kirawi 
- [x] boo_berry @bootradev 
- [x] bogster @vv9k 
- [x] everforest @CptPotato @WindSoilder 
- [ ] acme @two-six 
- [x] serika @VuiMuich 
- [ ] nord @raygervais @two-six 
- [x] dracula @samsartor 
- [ ] gruvbox @crodjer @jbaa 
- [x] autumn @getreu 
- [ ] ayu @enkodr 
- [x] rose_pine @raygervais @chunghha @atahabaki 
- [x] monokai @WindSoilder 
- [ ] solarized @kirawi @cossonleo 
- [x] spacebones @atog

AlexanderBrevig commented:
It's done @archseer
This PR contains all changes. I opted for a task list for keeping track of author feedback. There's really no intended aesthetic changes/choices made here though.

Let me know if you'd want side by side screenshots for any of them :)

WindSoilder commented:
Thank you! I'm ok with monokai and everforest changes

vlmutolo commented:
![2022-08-28_12-08-1661703023](https://user-images.githubusercontent.com/15202239/187083751-175caa4e-7344-4a3b-86b7-9493b6e7c88d.png)

The penumbra changes result in the command help text being unreadable on my machine.

What was the original issue the linter found with penumbra+? Maybe I can help resolve it.

AlexanderBrevig commented:
> ![2022-08-28_12-08-1661703023](https://user-images.githubusercontent.com/15202239/187083751-175caa4e-7344-4a3b-86b7-9493b6e7c88d.png)
> 
> The penumbra changes result in the command help text being unreadable on my machine.
> 
> What was the original issue the linter found with penumbra+? Maybe I can help resolve it.

Looking at it here it has sky for bg. That's wrong isn't it? Don't know what's happened, I'll fix it to shade later tonight. Thanks!

Penumbra only set fg for some fields that expects a bg. Most important were `ui.background` and `ui.help` and it seems somehow that a bug slipped through the cracks on this one where I used the original value (which in reality had no impact on the rendered theme and defaulted to terminal settings)

AlexanderBrevig commented:
Penumbra looks better now? 
![image](https://user-images.githubusercontent.com/721090/187090157-2f6b7ad2-7b2d-4938-9081-5f60c31ecccf.png)

nrdxp reviewed (approved):
Snazzy looks good :+1:

pickfire commented on runtime/themes/ayu_light.toml:
```diff
@@ -38,16 +38,17 @@
 "ui.popup" = { bg = "black" }
 "ui.window" = { fg = "dark_gray" }
-"ui.help" = { bg = "black" }
+"ui.help" = { fg="foreground", bg = "black" }
```
```suggestion
"ui.help" = { fg  = "foreground", bg = "black" }
```
If possible, please stick to the original format of the file, but I think you probably missed some.

AlexanderBrevig replied:
Thanks! I will update my linter to require fg and bg for help, and rerun.

pickfire commented on runtime/themes/ayu_light.toml:
```diff
@@ -38,16 +38,17 @@
 "ui.virtual.whitespace" = { fg = "dark_gray" }
 "ui.virtual.ruler" = { bg = "black" }
-"ui.menu" = { bg = "black" }
+"ui.menu" = { fg="foreground", bg = "black" }
```
```suggestion
"ui.menu" = { fg = "foreground", bg = "black" }
```

AlexanderBrevig commented:
Thanks for the heads up @pickfire , the linter is now updated to handle the ui better
```rust
        Rule::has_both("ui.menu".into()),
        Rule::has_both("ui.help".into()),
        Rule::has_bg("ui.popup".into()),
        Rule::has_either("ui.window".into()),
```

Updated a few themes that were not complete in this regard.

intarga commented:
I would have these instead:

```
"ui.linenr.selected" = { fg = "#ED5466", modifiers = ["bold"] }

"ui.cursor.primary" = { bg = "#ED5466", fg = "#FFFCFD", modifiers = ["bold"] }
"ui.cursor.match" = { bg = "#F3EAE9", fg = "#ED5466", modifiers = ["bold"] }
```

and if you don't mind adding another change while you're in there:

```
"ui.selection" = { bg = "#F3EAE9" }
```

Otherwise LGTM :)

VuiMuich commented:
Serika lgtm.
Thanks.

Generally I have the feeling though, that themes which don't use `modifiers = ["reverse"]` are handling the cursor being directly on a `ruler` not ideally, haven't checked yet if this is already known.

vlmutolo commented:
> Penumbra only set fg for some fields that expects a bg. 

Oops! Good catch.

> Penumbra looks better now?

Looks great. Only had the one minor comment.

ChrHorn commented:
> `Serika` themes lgtm. Thanks.
> 
> Generally I have the feeling though, that themes which don't use `modifiers = ["reverse"]` are handling the cursor being directly on a `ruler` not ideally, haven't checked yet if this is already known.

The ruler/cursor interaction seems to be a bug. Rulers should not overwrite the cursor styling.

AlexanderBrevig commented:
> `Serika` themes lgtm. Thanks.
> 
> Generally I have the feeling though, that themes which don't use `modifiers = ["reverse"]` are handling the cursor being directly on a `ruler` not ideally, haven't checked yet if this is already known.

PR for fix: https://github.com/helix-editor/helix/pull/3599

archseer commented:
By the way, can you rebase this PR? It seems to include duplicate commits from master

AlexanderBrevig commented:
Rebased.

pickfire reviewed (approved):
Looks good to me. I guess we don't really have to wait for all authors to get come back and tick this, those that wanted some changes can just submit a pull request separately later, but it's good to still notify them.

kirawi commented:
I'm not the author of the solarized themes by the way.
